### PR TITLE
Refactor apps suggestions

### PR DESCRIPTION
**Pull request by drazik**
The apps suggestions were crashing when there was a certain amount of transactions to analyze. There was two problems:

* A request to the DB was made for each transaction to find a potential corresponding app suggestion : now we fetch all existing suggestions in a single request
* The number of transactions listed in the relationships of a `io.cozy.apps.suggestions` document was not constrained, so it could grow very quickly. Writing big documents was  causing errors. We now save only the last 10 linked transactions

### CODE REVIEW

File: src/ducks/appSuggestions/AppSuggestion.js [hunk 1]
```diff
@@ -27,11 +12,13 @@ class AppSuggestion extends Document {
...
-      suggestion,
-      'relationships.transactions.data',
-      transactionRelationship
-    )
+
+    // We slice 9 existing transactions because we want to store only 10 items
+    // in the transactions relationship
+    suggestion.relationships.transactions.data = [
+      transactionRelationship,
+      ...suggestion.relationships.transactions.data.slice(0, 9)
```
Comment (ptbrowne) [inline src/ducks/appSuggestions/AppSuggestion.js]: Here the latest suggestion is first. Is it wanted ?

File: src/ducks/appSuggestions/services.spec.js [hunk 2]
```diff
@@ -8,28 +7,37 @@ describe('findSuggestionForTransaction', () => {
...
       }
-    })
+    ]
 
-    const suggestion = await findSuggestionForTransaction(transaction, brands)
+    const suggestion = await findSuggestionForTransaction(
+      transaction,
+      brands,
+      suggestions
+    )
```
Comment (ptbrowne) [inline src/ducks/appSuggestions/services.spec.js]: I could not see a test that was testing that we are using the originalSuggestion if it exists ? I think we need one if it does not exist.

File: src/ducks/appSuggestions/AppSuggestion.js (hunk 1 above)
Comment (drazik) [reply src/ducks/appSuggestions/AppSuggestion.js]: Yes it is wanted

File: src/ducks/appSuggestions/services.spec.js (hunk 2 above)
Comment (drazik) [reply src/ducks/appSuggestions/services.spec.js]: That's what this one is testing: we have an existing suggestion, and `findSuggestionForTransaction` uses it and updates it to add the transaction to the suggestion's relationships.

**Pull request merged**